networkx/algorithms/isomorphism/temporalisomorphvf2.py

```diff
@@ -0,0 +1,261 @@
+
+A brief discussion of the somewhat diverse current literature will be included here.
+
+References
+----------
+
+[1] Redmond, U. and Cunningham, P. Temporal subgraph isomorphism. In: The 2013 IEEE/ACM International Conference on Advances in Social Networks Analysis and Mining (ASONAM). Niagara Falls, Canada; 2013: pages 1451 - 1452. [65]
+
+[2] Redmond, U. and Cunningham, P. A temporal network analysis reveals the unprofitability of arbitrage in The Prosper Marketplace. In: Expert Systems with Applications; Pergamon; 2013: 40(9): pages 3715 - 3721. [64]
+
+
+Notes
+-----
+
+Handles directed and undirected graphs and graphs with parallel edges.
+
+"""
+
+#    This extension was originally coded by Ursula Redmond
+#    during postgraduate research at the Clique Research Cluster,
+#    and subsequently the Insight Centre for Data Analytics.
+#    Padraig Cunningham, principal investigator.
+#    School of Computer Science and Informatics, University College Dublin.
+
+from __future__ import absolute_import
+import networkx as nx
+from datetime import datetime, timedelta
+from .isomorphvf2 import GraphMatcher, DiGraphMatcher
+
+__all__ = ['TimeRespectingGraphMatcher',
+           'TimeRespectingDiGraphMatcher']
+
+
+class TimeRespectingGraphMatcher(GraphMatcher):
+
+    def __init__(self, G1, G2, temporal_attribute_name, delta):
+        """Initialize TimeRespectingGraphMatcher.
+
+        G1 and G2 should be nx.Graph or nx.MultiGraph instances.
+
+        Examples
+        --------
+        To create a TimeRespectingGraphMatcher which checks for syntactic and semantic feasibility:
+
+        >>> from networkx.algorithms import isomorphism
+        >>> G1 = nx.Graph(nx.path_graph(4, create_using=nx.Graph()))
+
+        >>> G2 = nx.Graph(nx.path_graph(4, create_using=nx.Graph()))
+
+        >>> GM = isomorphism.TimeRespectingGraphMatcher(G1, G2, 'date', timedelta(days=1))
+        """
+        self.temporal_attribute_name = temporal_attribute_name
+        self.delta = delta
+        super(TimeRespectingGraphMatcher, self).__init__(G1, G2)
+    
+    def one_hop(self, Gx, Gx_node, neighbors):
+        '''
+        Edges one hop out from a node in the mapping should be time-respecting w.r.t. each other.
+        '''
+        time_respecting = True
+        dates = []
+        for n in neighbors:
+            if type(Gx) == type(nx.Graph()): # Graph G[u][v] returns the data dictionary.
+                dates.append(Gx[Gx_node][n][self.temporal_attribute_name])
+            else: # MultiGraph G[u][v] returns a dictionary of key -> data dictionary.
+                for edge in Gx[Gx_node][n].values(): # Iterates all edges between node pair.
+                    dates.append(edge[self.temporal_attribute_name])
+        if any(x is None for x in dates):
+            raise ValueError('Datetime not supplied for at least one edge.')
+        dates.sort() # Small to large.
+        if 0 < len(dates) and not (dates[-1] - dates[0] <= self.delta):
```

Comment:
Simpler: `return not dates or max(dates) - min(dates) <= self.delta`, then you can remove the sort above.